### PR TITLE
Ignore secrets yaml on command line

### PR DESCRIPTION
**Pull request by cvwillegen**
# What does this implement/fix? 

When ``esphome`` is run from the CLI to update a subset of the configuration files, or all of them, the ``secrets.yaml`` file - when specified - yields an error. This makes it hard to do, for instance, ``esphome upload *yaml``. This code makes sure to skip the files ``secrets.yaml`` and ``secrets.yml`` when specified on the command line.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [X] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/__main__.py [hunk 1]
```diff
@@ -781,6 +777,20 @@ def run_esphome(argv):
         )
         return 1
 
+    if hasattr(args, "configuration"):
+        for secret in CONF_SECRETS:
+            if secret.casefold() in (
+                config.casefold() for config in args.configuration
+            ):
```
Comment (oxan) [inline esphome/__main__.py]: This check is too strict, we only want to exclude the literal filename `secrets.yaml`, not any path that contains that.

File: esphome/__main__.py [hunk 2]
```diff
@@ -781,6 +777,20 @@ def run_esphome(argv):
...
+                _LOGGER.warning(
+                    "%s was specified on the command line. Skipping...", secret
+                )
```
Comment (oxan) [inline esphome/__main__.py]: This is not very descriptive, might be nice to add a note why this file is being skipped.

File: esphome/__main__.py [hunk 3]
```diff
@@ -781,6 +777,20 @@ def run_esphome(argv):
...
+                args.configuration[:] = [
+                    conf
+                    for conf in args.configuration
+                    if conf.casefold() != secret.casefold()
+                ]
+
```
Comment (oxan) [inline esphome/__main__.py]: It's probably easier to do this in the for loop below, something like this:
```python
    for conf_path in args.configuration:
      if any(basename(conf_path) is x for x in CONF_SECRETS):
        _LOGGER.warning("skipping secrets file")
        continue
```
This also has the advantage that the `hasattr(args, "configuration")` is not necessary.

File: esphome/const.py [hunk 4]
```diff
@@ -591,6 +591,7 @@
 CONF_SDO_PIN = "sdo_pin"
 CONF_SECOND = "second"
 CONF_SECONDS = "seconds"
+CONF_SECRETS = ["secrets.yaml", "secrets.yml"]
```
Comment (oxan) [inline esphome/const.py]: This shouldn't be a `CONF_` constant, as those are for keys in the configuration YAML.

**Review by oxan — CHANGES_REQUESTED** [review]

File: esphome/__main__.py (hunk 1 above)
Comment (cvwillegen) [reply esphome/__main__.py]: Used your example code to skip files, thanks!

**Review by oxan — APPROVED** [review]

**Comment by oxan** [comment]
@jesserockz I think this is a good idea for usability, but it's technically a breaking change as you can no longer have a config named `secrets.yaml`. I don't think that's much of a problem, but thoughts?

**Review by jesserockz — APPROVED** [review]
That's fine. I would not expect people to have that anyway.

**Comment by oxan** [comment]
Thank you for your contribution!

**Comment by cvwillegen** [comment]
This has been an itch to scratch, but you're welcome! My second addition to esphome, both personal itches, but probably useful in the wild!

**Pull request merged**